pkg/pgmodel/cache/series_cache_test.go

```diff
@@ -52,3 +55,43 @@ func TestGenerateKey(t *testing.T) {
 	require.Equal(t, "test", metricName)
 	require.Equal(t, []byte("\x08\x00__name__\x04\x00test\x04\x00hell\x06\x00oworld\x05\x00hello\x05\x00world"), keyBuffer.Bytes())
 }
+
+func TestGrowSeriesCache(t *testing.T) {
+	testCases := []struct {
+		name         string
+		sleep        time.Duration
+		cacheGrowCnt int
+	}{
+		{
+			name:         "Grow criteria satisfied - we shoulnd't be evicting elements",
+			sleep:        time.Millisecond,
+			cacheGrowCnt: 1,
+		},
+		{
+			name:         "Growth criteria not satisfied - we can keep evicting old elements",
+			sleep:        time.Second * 2,
+			cacheGrowCnt: 0,
+		},
+	}
+
+	os.Setenv("IS_TEST", "true")
+	evictionMaxAge = time.Second // tweaking it to not wait too long
+	for _, tc := range testCases {
+		cache := NewSeriesCache(Config{SeriesCacheInitialSize: 100, SeriesCacheMemoryMaxBytes: DefaultConfig.SeriesCacheMemoryMaxBytes}, nil)
+		cacheGrowCounter := 0
+		for i := 0; i < 200; i++ {
+			cache.cache.Insert(i, i, 1)
+			if i%100 == 0 {
+				time.Sleep(tc.sleep)
```

Comment:
Another option that wouldn't require sleep is to add an attribute to the cache that would be a time provider, something like:

```
type timeProvider interface {
  now() int64 
```

That way you can control the clock for the tests. But that will require more changes.

Reply:
Sure. I just don't feel it's worth doing it